src/Makefile.am

```diff
@@ -661,6 +667,27 @@ bitcoin_wallet_SOURCES += bitcoin-wallet-res.rc
 endif
 #
 
+# bitcoin-util binary #
+bitcoin_util_SOURCES = bitcoin-util.cpp
+bitcoin_util_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES)
+bitcoin_util_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+bitcoin_util_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(LIBTOOL_APP_LDFLAGS)
+
+if TARGET_WINDOWS
+bitcoin_util_SOURCES += bitcoin-util-res.rc
+endif
+
+bitcoin_util_LDADD = \
+  $(LIBUNIVALUE) \
+  $(LIBBITCOIN_COMMON) \
+  $(LIBBITCOIN_UTIL) \
+  $(LIBBITCOIN_CONSENSUS) \
```

Comment:
"grind" needs libconsensus for both `primitives/block.h` for the block header definition and `arith_uint256.h` for converting nbits to a target and comparing the target to a sha256. Since libconsensus isn't linked into bitcoin-cli, that probably rules out `bitcoin-cli --grind=HEADER` as an alternative.

Making `bitcoin-tx -grind HEADER` work seems plausible; should look something like https://github.com/ajtowns/bitcoin/commits/202009-signet-generate-util2tx 

Long term I think `bitcoin-util` is a good idea, but it might be better to introduce it in its own PR, in which case putting the functionality in `bitcoin-tx` in the meantime seems plausible? Thoughts?

Reply:
`bitcoin-tx --grind` feels weird to me. Grinding is unrelated to transactions. Unless we decide `bitcoin-util` is not desired, I think introducing it here makes more sense, personally.